### PR TITLE
Setup Build Plugin 🐘 

### PR DESCRIPTION
**Pull request by jumaallan**
Story: https://app.shortcut.com/smileid/story/xxx

## Summary

A few sentences/bullet points about the changes

## Known Issues

Any shortcomings in your work. This may include corner cases not correctly handled or issues related
to but not within the scope of your PR. Design compromises should be discussed here if they were not
already discussed above.

## Test Instructions

Concise test instructions on how to verify that your feature works as intended. This should include
changes to the development environment (if applicable) and all commands needed to run your work.

## Screenshot

If applicable (e.g. UI changes), add screenshots to help explain your work.

### CODE REVIEW

File: build-logic/convention/src/main/kotlin/com/smileidentity/buildlogic/KotlinAndroid.kt [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+internal fun Project.configureKotlinAndroid(commonExtension: CommonExtension<*, *, *, *, *, *>) {
+    commonExtension.apply {
+        compileSdk = 36
+
+        defaultConfig {
+            minSdk = 21
+        }
+
+        compileOptions {
+            sourceCompatibility = JavaVersion.VERSION_11
```
Comment (wangerekaharun) [inline build-logic/convention/src/main/kotlin/com/smileidentity/buildlogic/KotlinAndroid.kt:21]: Lets set this to 17

File: build-logic/convention/src/main/kotlin/com/smileidentity/buildlogic/KotlinAndroid.kt [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    // Treat all Kotlin warnings as errors (disabled by default)
+    // Override by setting warningsAsErrors=true in your ~/.gradle/gradle.properties
+    val warningsAsErrors = providers.gradleProperty("warningsAsErrors").map {
+        it.toBoolean()
+    }.orElse(false)
+    when (this) {
+        is KotlinAndroidProjectExtension -> compilerOptions
+        else -> TODO("Unsupported project extension $this ${T::class}")
+    }.apply {
+        jvmTarget = JvmTarget.JVM_11
```
Comment (wangerekaharun) [inline build-logic/convention/src/main/kotlin/com/smileidentity/buildlogic/KotlinAndroid.kt:39]: Let's set this to 17

File: build-logic/convention/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import org.jetbrains.kotlin.gradle.dsl.JvmTarget
+
+plugins {
+    `kotlin-dsl`
+}
+
+group = "com.smileidentity.buildlogic"
```
Comment (wangerekaharun) [inline build-logic/convention/build.gradle.kts:7]: Minor nit, do we go with smileid or smileidentity?

Comment (wangerekaharun) [reply build-logic/convention/build.gradle.kts:7]: Yeah, also the plugins can reflect this
